lib/internal/repl.js

```diff
@@ -84,18 +90,26 @@ function setupHistory(repl, historyPath, ready) {
     if (err) {
       return ready(err);
     }
-
-    if (data) {
+    if (data && data[0] === '['
+        // Check the last two characters in case there is a trailing newline
+        && /\]/.test(data.substr(-2))) {
       try {
         repl.history = JSON.parse(data);
         if (!Array.isArray(repl.history)) {
           throw new Error('Expected array, got ' + typeof repl.history);
         }
         repl.history.slice(-repl.historySize);
+
+        // update the history file to use plain text
+        const historyData = repl.history.join(os.EOL);
+        writing = true;
+        fs.write(repl._historyHandle, historyData, 'utf8', onwritten);
```

Comment:
Hmm, is that something we can fix in 2.x? I.e. make it at least not barf? Or even be able to interpret plain-text but not default to it?